specification/network/resource-manager/Microsoft.Network/2017-08-01/examples/UsageList.json

```diff
@@ -9,103 +9,254 @@
       "body": {
         "value": [
           {
-            "unit": "Count",
-            "currentValue": 47,
-            "limit": 100,
+            "currentValue": 8.0,
```

Comment:
I noticed these are now decimal values.  

in `usage.json` -- these values are listed as `int64` format and you're sending in non-integer values.

I don't think that this is OK -- if the values are coming back as non-integers, some deserializers will probably fail. Is there a service change?

``` json
   "Usage": {
      "properties": {
        "unit": {
          "type": "string",
          "description": "An enum describing the unit of measurement.",
          "enum": [
            "Count"
          ],
          "x-ms-enum": {
            "name": "UsageUnit",
            "modelAsString": true
          }
        },
        "currentValue": {
          "type": "integer",
          "format": "int64",
          "description": "The current value of the usage."
        },
        "limit": {
          "type": "integer",
          "format": "int64",
          "description": "The limit of usage."
        },
        "name": {
          "$ref": "#/definitions/UsageName",
          "description": "The name of the type of usage."
        }
      },
```

Reply:
we will not be sending non-integer values. sending non-integer values does not make sense for this api. 

To be honest, the service uses this object Microsoft.Azure.Insights.Models.UsageMetric to populate the response and property definitions are as follows
private double _currentValue;
    private string _id;
    private double _limit;
    private LocalizableString _name;
    private string _nextResetTime;
    private TimeSpan? _quotaPeriod;
    private string _unit;

Even though it says double currentValue, it doesnt make sense to introduce a breaking change on our client (from int64 to double).

Reply:
The trouble is, some deserializers may not properly handle deserializing values like `50.0` into an integer field, even if the value is actually a whole number. If your service is returning number values with dots in them, that may not be good for some clients.

I'm more concerned with what the service is returning...

Reply:
at this point, i dont know what we can do here. you might have to take it up with the insight people. All usages implementations use that i believe.

Reply:
i thikn the right thing to do would be to convert the clients to double but that would be a breaking change :( . do you think we ought to do that. does this warrant a breaking change for the client?

Reply:
I think we just stepped into a lose/lose situation.

If the service has been returning floating point numbers all along (even if they are whole numbers) -- that means the clients are broken already. 

Which means, we're better off correcting the specification with the correct types and publishing a new version of the package, even if it means that it's a minor breaking change. 

Given that the other things in this PR are corrections which may introduce minor breaking effects,  my preference is that let's do the right thing and fix the spec and generate the new code.

Reply:
i agree with you but the not the timing. With ignite close by, i dont want to do this now. can we do this post ignite? this would require changes in PS as well.

Reply:
@fearthecowboy, i have opened an issue for this that we will get to after ignite: https://github.com/Azure/azure-rest-api-specs/issues/1624